core/net/rpl/rpl-ext-header.c

```diff
@@ -378,20 +388,20 @@ insert_srh_header()
     cmpri = MIN(cmpri, count_matching_bytes(&node_addr, &UIP_IP_BUF->destipaddr, 16));
     cmpre = cmpri;
 
-    PRINTF("RPL: SRH Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
+    if(node->parent != root_node) {
+      PRINTF("RPL: SRH Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    } else if(((DEBUG) & DEBUG_PRINT)) {
+      rpl_ns_get_node_global_addr(&node_addr, node);
+      PRINTF("RPL: SRH Next Hop ");
+      PRINT6ADDR(&node_addr);
+      PRINTF("\n");
+    }
     node = node->parent;
     path_len++;
   }
 
-  if(((DEBUG) & DEBUG_PRINT) && node != NULL) {
-    rpl_ns_get_node_global_addr(&node_addr, node);
-    PRINTF("RPL: SRH Next Hop ");
-    PRINT6ADDR(&node_addr);
-    PRINTF("\n");
-  }
```

Comment:
I don't have any particular opinion on that, actually. I just kept it as it was. :-)

I thought you did not want to call `rpl_ns_get_node_global_addr()` unless `DEBUG_PRINT` is defined. If it is not the case, it would be better to modify the part with a simple `else` statement.